packages/components/src/components/primitives/Link/Link.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { FC } from 'react';
+import { Box, Link as ChakraLink, useStyleConfig } from '@chakra-ui/react';
+
+import { LinkProps } from './types';
+import { marginSizes, linkIconSizes } from './types';
+
+export const Link: FC<LinkProps> = ({
+  children,
+  icon,
+  iconPosition = 'left',
+  size = 'm',
+  ...props
+}) => {
+  const { textStyle } = useStyleConfig('Link', {
+    size: size,
+  });
+
+  const calculateMargin = () =>
+    iconPosition === 'left'
+      ? { marginRight: marginSizes[size] }
+      : iconPosition === 'right'
+      ? { marginLeft: marginSizes[size] }
+      : {};
+
+  const Icon: React.FC = () =>
+    React.cloneElement(icon, {
+      boxSize: linkIconSizes[size],
+    });
```

Comment:
Can't we use the `Icon` and pass the `boxSize` property without cloning the element?